addons/mail/models/mail_render_mixin.py

```diff
@@ -269,8 +276,12 @@ def _render_template_qweb(self, template_src, model, res_ids,
         for record in self.env[model].browse(res_ids):
             variables['object'] = record
             try:
-                render_result = self.env['ir.qweb']._render(html.fragment_fromstring(
-                    template_src, create_parent='div'), variables, raise_on_code=is_restricted)
+                self.env.context = dict(self.env.context, preserve_comments=True)
+                render_result = self.env['ir.qweb']._render(
```

Comment:
Seems we are now "always" giving preserve_comments ? Was not the case previously, normally caller updates the context and we should not update it again here, unless I misunderstood something ?